Library/Homebrew/extend/cachable.rb

```diff
@@ -7,9 +7,10 @@ def cache
     @cache ||= T.let({}, T.nilable(T::Hash[T.untyped, T.untyped]))
   end
 
+  # NOTE: We overwrite here instead of using `Hash#clear` to handle frozen hashes.
   sig { void }
   def clear_cache
-    cache.clear
+    overwrite_cache!({})
```

Comment:
The internal JSON v3 format gets parsed into a frozen hash with `JSON.parse(string, frozen: true)` and then gets moved into the cache with `Cachable#overwrite_cache!` so the cache is frozen in that case.

Reply:
I'm not sure if there are any real downsides to creating a new hash here instead of using `Hash#clear`.